src/xcode/ENA/ENA/Source/Developer Menu/Features/DMKeysViewController.swift

```diff
@@ -0,0 +1,207 @@
+//
+// Corona-Warn-App
+//
+// SAP SE and all other contributors
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+#if !RELEASE
+
+import ExposureNotification
+import UIKit
+
+/// The root view controller of the developer menu.
+final class DMKeysViewController: UITableViewController {
+	// MARK: Creating a developer menu view controller
+
+	init(
+		client: Client,
+		store: Store,
+		exposureManager: ExposureManager
+	) {
+		self.client = client
+		self.store = store
+		self.exposureManager = exposureManager
+		super.init(style: .plain)
+		title = "🔑 Diagnosis Keys 🔑"
+	}
+
+	@available(*, unavailable)
+	required init?(coder _: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: Properties
+
+	private let client: Client
+	private let store: Store
+	private let exposureManager: ExposureManager
+	private var keys = [SAP_TemporaryExposureKey]() {
+		didSet {
+			keys = self.keys.sorted()
+		}
+	}
+
+	// MARK: UIViewController
+
+	override func viewWillAppear(_ animated: Bool) {
+		navigationController?.setToolbarHidden(false, animated: animated)
+		setToolbarItems(
+			[
+				UIBarButtonItem(
+					title: "Diag. Keys",
+					style: .plain,
+					target: self,
+					action: #selector(refreshKeys)
+				),
+				UIBarButtonItem(
+					barButtonSystemItem: .flexibleSpace,
+					target: nil,
+					action: nil
+				),
+				UIBarButtonItem(
+					title: "Test Keys",
+					style: .plain,
+					target: self,
+					action: #selector(generateTestKeysAndRefresh)
+				),
+				UIBarButtonItem(
+					barButtonSystemItem: .flexibleSpace,
+					target: nil,
+					action: nil
+				),
+				UIBarButtonItem(
+					title: "Export",
+					style: .plain,
+					target: self,
+					action: #selector(exportKeys)
+				)
+			],
+			animated: animated
+		)
+		super.viewWillAppear(animated)
+	}
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		tableView.register(DMKeyCell.self, forCellReuseIdentifier: DMKeyCell.reuseIdentifier)
+	}
+
+	// MARK: Working with the Keys
+	@objc
+	private func refreshKeys() {
+		keys = []
+		tableView.reloadData()
+		exposureManager.accessDiagnosisKeys { keys, _ in
+			guard let keys = keys else {
+				logError(message: "No keys retrieved in developer menu")
```

Comment:
I hope that this will never fail in the first place. If it does you should see it in the new error log within the developer menu. :) In a perfect world I also would love to see an alert here but at some point I decided to not make the developer menu any better - otherwise the PR would be even bigger.